pkg/registries/adapters/local_openshift_adapter.go

```diff
@@ -0,0 +1,125 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Red Hat trademarks are not licensed under Apache License, Version 2.
+// No permission is granted to use or replicate Red Hat trademarks that
+// are incorporated in this software or its documentation.
+//
+
+package adapters
+
+import (
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/apb"
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+	yaml "gopkg.in/yaml.v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+const localOpenShiftName = "openshift-registry"
+
+// LocalOpenShiftAdapter - Docker Hub Adapter
+type LocalOpenShiftAdapter struct {
+	Config Configuration
+	Log    *logging.Logger
+}
+
+// RegistryName - Retrieve the registry name
+func (r LocalOpenShiftAdapter) RegistryName() string {
+	return localOpenShiftName
+}
+
+// GetImageNames - retrieve the images
+func (r LocalOpenShiftAdapter) GetImageNames() ([]string, error) {
+	r.Log.Debug("LocalOpenShiftAdapter::GetImageNames")
+	r.Log.Debug("BundleSpecLabel: %s", BundleSpecLabel)
+
+	openshiftClient, err := clients.Openshift(r.Log)
+	images, err := openshiftClient.ListRegistryImages(r.Log)
+	if err != nil {
+		r.Log.Errorf("Failed to load registry images")
+		return nil, err
+	}
+
+	return images, nil
+}
+
+// FetchSpecs - retrieve the spec for the image names.
+func (r LocalOpenShiftAdapter) FetchSpecs(imageNames []string) ([]*apb.Spec, error) {
+	r.Log.Debug("LocalOpenShiftAdapter::FetchSpecs")
+	r.Log.Debugf("HERE %v", r.Config.Namespaces)
+	specList := []*apb.Spec{}
+	registryIP, err := r.getServiceIP("docker-registry", "default")
+
+	openshiftClient, err := clients.Openshift(r.Log)
+	fqImages, err := openshiftClient.ConvertRegistryImagesToSpecs(r.Log, imageNames)
+	if err != nil {
+		r.Log.Errorf("Failed to load registry images")
+		return nil, err
+	}
+
+	for _, image := range fqImages {
+		spec, err := r.loadSpec(image.DecodedSpec)
+		if err != nil {
+			r.Log.Errorf("Failed to load image spec")
+			continue
+		}
+		if strings.HasPrefix(image.Name, registryIP) {
+			// Image has proper registry IP prefix
+			spec.Image = image.Name
+			namespace := strings.Split(image.Name, "/")[1]
+			for _, ns := range r.Config.Namespaces {
+				if ns == namespace {
+					r.Log.Debugf("Image [%v] is in configured namespace [%v]. Adding to SpecList.", image.Name, ns)
+					specList = append(specList, spec)
+				}
+			}
+		} else {
+			r.Log.Debugf("Image does not have proper registry IP prefix. Something went wrong.")
+		}
+	}
+
+	return specList, nil
+}
+
+func (r LocalOpenShiftAdapter) loadSpec(yamlSpec []byte) (*apb.Spec, error) {
+	r.Log.Debug("LocalOpenShiftAdapter::LoadSpec")
+	var err error
```

Comment:
weird to declare a var when you can easily do `err := yaml.Unmarshal` on line 103.